airflow/providers/amazon/aws/operators/datasync.py

```diff
@@ -62,60 +80,102 @@ class AWSDataSyncCreateTaskOperator(BaseOperator):
+        "update_task_kwargs",
+    )
+    ui_color = "#44b5e2"
 
     @apply_defaults
     def __init__(
         self,
-        aws_conn_id='aws_default',
+        aws_conn_id="aws_default",
+        wait_interval_seconds=5,
+        task_arn=None,
         source_location_uri=None,
         destination_location_uri=None,
-        case_sensitive_location_search=True,
+        location_search_case_sensitive=True,
+        location_search_ignore_trailing_slash=True,
+        choose_task_callable=None,
+        choose_location_callable=None,
         create_task_kwargs=None,
         create_source_location_kwargs=None,
         create_destination_location_kwargs=None,
+        update_task_kwargs=None,
+        delete_task_after_execution=False,
         *args,
         **kwargs
     ):
         super().__init__(*args, **kwargs)
 
         # Assignments
         self.aws_conn_id = aws_conn_id
+        self.wait_interval_seconds = wait_interval_seconds
+
+        self.task_arn = task_arn
+
         self.source_location_uri = source_location_uri
         self.destination_location_uri = destination_location_uri
-        self.case_sensitive_location_search = case_sensitive_location_search
-        if create_task_kwargs:
-            self.create_task_kwargs = create_task_kwargs
-        else:
-            self.create_task_kwargs = dict()
+
+        self.location_search_case_sensitive = location_search_case_sensitive
+        self.location_search_ignore_trailing_slash = (
+            location_search_ignore_trailing_slash
+        )
+
+        self.choose_task_callable = choose_task_callable
+        self.choose_location_callable = choose_location_callable
+
+        self.create_task_kwargs = create_task_kwargs if create_task_kwargs else dict()
+        self.create_source_location_kwargs = dict()
         if create_source_location_kwargs:
             self.create_source_location_kwargs = create_source_location_kwargs
-        else:
-            self.create_source_location_kwargs = dict()
+        self.create_destination_location_kwargs = dict()
         if create_destination_location_kwargs:
             self.create_destination_location_kwargs = create_destination_location_kwargs
-        else:
-            self.create_destination_location_kwargs = dict()
+
+        self.update_task_kwargs = update_task_kwargs if update_task_kwargs else dict()
+        self.delete_task_after_execution = delete_task_after_execution
 
         # Validations
-        if not (self.source_location_uri and self.destination_location_uri):
+        valid = False
+        if self.task_arn:
+            valid = True
+        if self.source_location_uri and self.destination_location_uri:
+            valid = True
+        if not valid:
             raise AirflowException(
-                'Specify both source_location_uri and destination_location_uri')
+                "Specify task_arn or both source_location_uri and destination_location_uri"
```

Comment:
Might be good to f-string the variables here so that you can see which one was missing.

Reply:
Thanks, done :)